src/transaction-flow/input/SelectPrimaryName/SelectPrimaryName.test.tsx

```diff
@@ -0,0 +1,153 @@
+import { mockFunction, render, screen, userEvent, waitFor } from '@app/test-utils'
+
+import { useAvailablePrimaryNamesForAddress } from '@app/hooks/useAvailablePrimaryNamesForAddress'
+import { useEns } from '@app/utils/EnsProvider'
+
+import SelectPrimaryName from './SelectPrimaryName-flow'
+
+jest.mock('@app/hooks/useChainId', () => ({
+  useChainId: jest.fn().mockReturnValue(1),
+}))
+
+const mockInvalidateQueries = jest.fn()
+jest.mock('wagmi', () =>
+  jest.fn().mockReturnValue({
+    useQueryClient: jest.fn().mockReturnValue({
+      invalidateQueries: mockInvalidateQueries,
+    }),
+  }),
+)
+jest.mock('@app/components/@atoms/NameDetailItem/TaggedNameItem', () => ({
+  TaggedNameItem: ({ name, ...props }: any) => <div {...props}>{name}</div>,
+}))
+
+jest.mock('@app/hooks/useAvailablePrimaryNamesForAddress')
+jest.mock('@app/utils/EnsProvider')
+
+const mockUseEns = mockFunction(useEns)
+const mockUseAvailablePrimaryNamesForAddress = mockFunction(useAvailablePrimaryNamesForAddress)
+
+const makeName = (index: number) => ({
+  name: `test${index}.eth`,
+  id: `0x${index}`,
+})
+
+describe('SelectPrimaryName', () => {
```

Comment:
tests for `onSubmit()` variants? (e.g. 1 step set name, 2 step no matching eth, 3 step no valid resolver)